### DISCUSSION TITLE
_app. componentDidCatch is deprecated in upcoming v9.3.7 - Is the Sentry example still valid?

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/next.js#12562] [Help] [answered]
I've noticed the `componentDidCatch` method is now deprecated. 

https://github.com/zeit/next.js/blob/v9.3.7-canary.12/packages/next/pages/_app.tsx#L38

But, it's still used in the with-sentry example:
https://github.com/zeit/next.js/blob/18a9c7e371efc4c487f9c3599c3211ce30009d6c/examples/with-sentry/pages/_app.js#L50

Should we stop using it? I'm wondering because I'm also using Sentry with the following code:

```tsx
  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
    // eslint-disable-next-line no-console
    console.debug('_app.componentDidCatch - Unexpected error caught', error, errorInfo);

    Sentry.withScope((scope) => {
      Object.keys(errorInfo).forEach((key) => {
        scope.setExtra(key, errorInfo[key]);
      });

      Sentry.captureException(error);
    });

    // This is needed to render errors correctly in development / production
    super.componentDidCatch(error, errorInfo);
  }
```

Is the above way outdated? 
Do you recommend a new way of catching errors as of Next.js 9.3.6?

**Comment by timneutkens** [comment, marked as answer]
It's actually not using that method, using componentDidCatch is fine in any component. What was deprecated is having to call `super.componentDidCatch` (calling the built-in _app componentDidCatch) as that does not do anything special anymore.

So in case of your example you can remove `super.componentDidCatch` and replace it with `throw error` as that makes it bubble up the error again.

**Reply by Vadorequest** [reply]
Thanks @timneutkens, I've actually removed `componentDidCatch` in _app altogether, I don't see the point anymore. I don't remember why it was necessary in the first place. Errors in `_app` are properly caught and display the `_error` page as expected, so it doesn't seem necessary to keep it around.

Unless I've missed something?

**Reply by timneutkens** [reply]
Correct! We just catch them and render _error indeed 👍

**Reply by Vadorequest** [reply]
Awesome, this simplifies things a bit.

I have other error-related questions at https://github.com/zeit/next.js/discussions/12913 (with focus on SSG/hybrid rendering and concerns), if ever you find some time.

I'm also rewriting a v9.2 app into a v9.4 with SSG first-class support at https://github.com/UnlyEd/next-right-now/tree/v1-hyb-mst-aptd-gcms-lcz-sty, I'd love if someone from the Next.js core team could review it and give us some feedback on it (perfs, best practices, etc.)

Thank you for this very great framework. <3
